### PR TITLE
Feature/freeze scheduler

### PR DESCRIPTION
**Pull request by AgaMiko**
Add a basic freeze scheduler that allows planning which model-named parameters (weights) can be frozen and unfrozen during the training. Frozen weights mean that those weights are not trained. The scheduler params are passed in models_config and can be user-defined. 
The scheduler can be used every epoch or every step.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -200,7 +198,16 @@ def main(args):
         "transformer_name": "sign_language_transformer",
         "model_save_dir": args.save,
         "neptune": args.neptune,
-        "device": "cuda:0" if args.gpu > 0 else "cpu",    
+        "device": "cuda:0" if args.gpu > 0 else "cpu",
+        "freeze_scheduler": {
+            "model_params": {
```
Comment (AgaMiko) [inline train.py:203]: example parameters

File: README.md [hunk 2]
```diff
@@ -87,8 +115,38 @@ Run with multiple datasets (list datasets paths separated with spaces)
 
 `python3 train.py --data "/dih4/dih4_2/hearai/data/frames/pjm" "/dih4/dih4_2/hearai/data/frames/basic_lexicon" --epochs 100 --lr 1e-4 --classification-mode "hamnosys" --neptune --num_segments 16 --b 4 --workers 0 --gpu 1`
 
+# Training tips & tricks
+
+### Schudule model freezing
+If you use a pretrained `feature_extractor` but training `transfomer` from sctratch its worth to freeze weights of feature_extractor at first, and train the model with higher learning rate. With `freeze_scheduler` you can quickly prepare your freezing configuration. Freeze pattern can be used with any named model parameter. To use it, first select layers which you want to freeze, and add named parameters to `model_params` in the config file. For instance, it can be used with `feature_extractor` and `transformer`. When the value is set to `True` then the layer will be freezed after first freeze_scheduler execution.
```
Comment (AgaMiko) [inline README.md:121]: explanation in README.md

File: models/model.py [hunk 3]
```diff
@@ -189,3 +205,64 @@ def optimizer_step(
         optimizer.step(closure=optimizer_closure)
         if self.run:
             self.run["params/lr"].log(optimizer.param_groups[0]["lr"])
+
+    def configure_freeze_scheduler(self):
+        ### TO-DO check if all params are correctly set
```
Comment (AgaMiko) [inline models/model.py:210]: set scheduler (naming style like for optimizers and weight schedulers)

File: models/model.py [hunk 4]
```diff
@@ -189,3 +205,64 @@ def optimizer_step(
...
+        # e.g. check if all lists are the same length
+        # check if values are bools
+        self.freeze_scheduler["current_pattern"] = 0
+        self.freeze_scheduler["current_counter"] = 0
+        self.freeze_step()
+
+    def freeze_step(self):
+        ### TO- DO
```
Comment (AgaMiko) [inline models/model.py:218]: single freeze step  (naming style like for optimizers and weight schedulers)

**Review by mariaogryczak — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (natcz) [reply README.md:121]: it should be 'schedule' not 'schudule'

**Review by polquant — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (AgaMiko) [reply README.md:121]: fixed
